src/module.c

```
@@ -11201,35 +11210,54 @@ long long getModuleNumericConfig(ModuleConfig *module_config) {
+            config_pair = sdssplitargs(supplied_config, &count);
+            if (!(strcasecmp(config_name, config_pair[0]))) {
+                if (!performModuleConfigSetFromName(config_pair[0], config_pair[1], &err)) {
+                    serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", config_pair[0], err);
+                    sdsfreesplitres(config_pair, count);
+                    sdsfree(config_name);
+                    listEmpty(server.module_configs_queue);
+                    return REDISMODULE_ERR;
+                }
+                config_supplied = 1;
+                sdsfreesplitres(config_pair, count);
+                listDelNode(server.module_configs_queue, ln_supplied);
+                break;
+            }
             sdsfreesplitres(config_pair, count);
-            listEmpty(server.module_configs_queue);
-            return REDISMODULE_ERR;
         }
-        if (!performConfigSetFromName(config_pair[0], config_pair[1], &err)) {
-            serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", config_pair[0], err);
-            sdsfreesplitres(config_pair, count);
-            listEmpty(server.module_configs_queue);
-            return REDISMODULE_ERR;
+        
+        if (!config_supplied) {
+            if (!performModuleConfigInitFromName(config_name, &err)) {
+                serverLog(LL_WARNING, "Issue attempting to set default value of configuration %s : %s", module_config->name, err);
+                listEmpty(server.module_configs_queue);
+                sdsfree(config_name);
+                return REDISMODULE_ERR;
+            }
         }
-        sdsfreesplitres(config_pair, count);
-        listDelNode(server.module_configs_queue, ln);
+        sdsfree(config_name);
+        config_supplied = 0;
+    }
+    if (is_loadex && listLength(server.module_configs_queue)) {
+        listEmpty(server.module_configs_queue);
+        return REDISMODULE_ERR;
```

maybe we want to print a log message with something about the contents of the list? e.g. at least one entry.
p.s. considering we do that only for LOADEX, maybe it's a hint to indicate that this shouldn't be here..
i.e. for the case of server startup, we do that after done loading all modules, and for the case of LOADEX we do that in the LOADEX command?
this will also mean that this function doesn't need to take the `is_loadex` argument (which is a win IMHO)

---

Should absolutely have a log message, this is included only for loadex due to the situation where multiple modules with configs are loaded from a .conf file. The reason its currently located here is for a rather esoteric edge case where a string config is initialized, and theres a bad argument to loadex. If we do the length check after onLoad is completed, the module author never gets the chance to free the string.

---

you mean that we can return an error to RM_LoadConfigs in that case?
i'm not sure that's right.. arguably the module did everything right, and it's just that the user passed an invalid argument to LOADEX, so LOADEX could return an error, and possibly unload the module.
or if we can't / don't wan to unload the module, maybe it's ok to return an error that indicates the module was loaded but the configuration failed...
also, i suppose logging a message to the log file with the name of the invalid config is not enough.. maybe we better have that as part of the reply?